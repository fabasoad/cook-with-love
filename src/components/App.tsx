import * as React from 'react'
import { connect } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'

import BackToTopButton from './BackToTopButton'
import FooterMain from './Footer/FooterMain'
import Header from './Header'
import LanguagesMain from './Languages/LanguagesMain'
import MenuMain from './Menu/MenuMain'
import RangeMain from './Range/RangeMain'
import { Locale, AppActions } from '../store/locale/types'
import { AppState } from '../store/configureStore'

interface AppProps {
}

type Props = AppProps & LinkDispatchProps & LinkStateProps

export function App(props: Props) {
  return (
    <div className={`font-regular font-${props.locale.code === 'jp' ? '' : 'non-'}jp`}>
      <BackToTopButton />
      <Header />
      <MenuMain />
      <RangeMain />
      <LanguagesMain />
      <FooterMain />
    </div>
  )
}

interface LinkStateProps {
  locale: Locale
}

interface LinkDispatchProps {
}

const mapStateToProps = (
  state: AppState,
  ownProps: AppProps
): LinkStateProps => ({
  locale: state.locale
})

const mapDispatchToProps = (
  dispatch: ThunkDispatch<any, any, AppActions>,
  ownProps: AppProps
): LinkDispatchProps => ({
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
