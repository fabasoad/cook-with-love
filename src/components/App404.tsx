import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { connect } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'
import { Locale, AppActions } from '../store/locale/types'
import { AppState } from '../store/configureStore'

interface App404Props {
}

type Props = App404Props & LinkDispatchProps & LinkStateProps

export function App404(props: Props) {
  const { t } = useTranslation()


  return (
    <div className={`font-regular font-${props.locale.code === 'jp' ? '' : 'non-'}jp`}>
      <p>{t('cook-with-love-404-text-1')}</p>
    </div>
  )
}

interface LinkStateProps {
  locale: Locale
}

interface LinkDispatchProps {
}

const mapStateToProps = (
  state: AppState,
  ownProps: App404Props
): LinkStateProps => ({
  locale: state.locale
})

const mapDispatchToProps = (
  dispatch: ThunkDispatch<any, any, AppActions>,
  ownProps: App404Props
): LinkDispatchProps => ({
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App404)
