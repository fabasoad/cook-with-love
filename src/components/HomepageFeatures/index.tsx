import React from 'react';
import clsx from 'clsx';
import styles from './styles.module.css';

type FeatureItem = {
  title: string;
  description: JSX.Element;
};

const FeatureList: FeatureItem[] = [
  {
    title: 'Cakes',
    description: (
      <>
        Welcome to a world of sweetness and indulgence at our online cake shop.
        Explore our divine assortment of cakes, lovingly crafted to satisfy your
        sweet cravings. From luscious chocolate decadence to fruity delights, our
        cakes are designed to add a touch of magic to your celebrations or simply
        brighten your day. Order now and let the flavors of our exquisite cakes
        transport you to a world of irresistible treats.
      </>
    ),
  },
  {
    title: 'Macarons',
    description: (
      <>
        Discover the epitome of elegance and flavor with our exquisite macarons.
        Delicately crafted, each bite of our macarons is a burst of perfection,
        a symphony of taste. Indulge in a world of vibrant colors and delightful
        fillings, making every moment a sweet celebration. Shop now and experience
        the art of macaron magic with us.
      </>
    ),
  },
  {
    title: 'Marshmallows',
    description: (
      <>
        Treat yourself to the fluffy, melt-in-your-mouth goodness of our gourmet
        marshmallows. Handcrafted with care, our marshmallows are the epitome of
        sweet, pillowy delight. From classic vanilla to innovative flavor combinations,
        they're perfect for toasting, s'mores, or simply enjoying as a sweet,
        cozy treat. Shop now and experience the joy of marshmallow indulgence
        like never before.
      </>
    ),
  },
];

function Feature({title, description}: FeatureItem) {
  return (
    <div className={clsx('col col--4')}>
      <div className="text--center padding-horiz--md">
        <h3>{title}</h3>
        <p>{description}</p>
      </div>
    </div>
  );
}

export default function HomepageFeatures(): JSX.Element {
  return (
    <section className={styles.features}>
      <div className="container">
        <div className="row">
          {FeatureList.map((props, idx) => (
            <Feature key={idx} {...props} />
          ))}
        </div>
      </div>
    </section>
  );
}
