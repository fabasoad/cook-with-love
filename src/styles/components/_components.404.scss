html, body {
  height: 96%;
}

body {
  background-color: $dark-grey;
}

.container {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  >div {
    color: $white;
    text-align: center;
  }
}
