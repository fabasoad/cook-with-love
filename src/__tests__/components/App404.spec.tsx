/// <reference types="jest" />
import * as React from 'react'
import { shallow } from 'enzyme'
import { App404 } from '../../components/App404'

new Array<string>('jp', 'gb').forEach((l) =>
  test(`should render App404 with ${l} locale correctly`, () =>
    expect(shallow(<App404 locale={{ code: l, title: '' }} />)).toMatchSnapshot()
  )
)
