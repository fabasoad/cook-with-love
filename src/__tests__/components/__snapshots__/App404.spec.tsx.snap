// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App404 with gb locale correctly 1`] = `
<div
  className="font-regular font-non-jp"
>
  <p>
    cook-with-love-404-text-1
  </p>
</div>
`;

exports[`should render App404 with jp locale correctly 1`] = `
<div
  className="font-regular font-jp"
>
  <p>
    cook-with-love-404-text-1
  </p>
</div>
`;
